src/browser/dom/document.zig

```diff
@@ -319,9 +319,68 @@ pub const Document = struct {
         log.debug(.web_api, "not implemented", .{ .feature = "Document hasFocus" });
         return true;
     }
+
+    pub fn _open(_: *parser.Document, page: *Page) !*parser.DocumentHTML {
+        if (page.open) {
+            return page.window.document;
+        }
+
+        // This implementation is invalid.
+        // According to MDN, we should cleanup registered listeners.
+        // So we sould cleanup previous DOM memory.
+        // But this implementation is more simple for now.
+        const html_doc = try parser.documentHTMLParseFromStr("");
+        try page.setDocument(html_doc);
+        page.open = true;
+
+        return page.window.document;
+    }
+
+    pub fn _close(_: *parser.Document, page: *Page) !void {
+        page.open = false;
+    }
+
+    pub fn _write(self: *parser.Document, str: []const u8, page: *Page) !void {
+        if (!page.open) {
```

Comment:
I think you can call ` _ = try _open(self, page)` directly here (without the `if` check), since it does that `if` check for you. Seems better for `write` not to know about `page.open` if possible.